src/app/view/components/container-selector/container-selector.component.spec.ts

```diff
@@ -0,0 +1,147 @@
+import { ContainerSelectorComponent } from './container-selector.component';
+
+describe('ContainerSelectorComponent', () => {
+    let comp: ContainerSelectorComponent;
+    let fixture: ComponentFixture<ContainerSelectorComponent>;
+    let de: DebugElement;
+    let el: HTMLElement;
+
+    beforeEach(async(() => {
+        const messageServiceMock = new MockMessageService({
+            addcontainer: 'Add a Container'
+        });
+
+        DOTTestBed.configureTestingModule({
+            declarations: [ContainerSelectorComponent],
+            imports: [ SearchableDropDownModule ],
+            providers: [
+                { provide: MessageService, useValue: messageServiceMock },
+                IframeOverlayService,
+                PaginatorService
+            ]
+        });
+
+        fixture = DOTTestBed.createComponent(ContainerSelectorComponent);
+        comp = fixture.componentInstance;
+        de = fixture.debugElement;
+        el = de.nativeElement;
+
+        this.searchableDropdownComponent = de.query(By.css('searchable-dropdown')).componentInstance;
+
+        this.containers = [
+            {
+                categoryId: '427c47a4-c380-439f-a6d0-97d81deed57e',
+                deleted: false,
+                friendlyName: 'Friendly Container name',
+                identifier: '427c47a4-c380-439f',
+                name: 'Container 1',
+                type: 'Container'
+            },
+            {
+                categoryId: '40204d-c380-439f-a6d0-97d8sdeed57e',
+                deleted: false,
+                friendlyName: 'Friendly Container2 name',
+                identifier: '427c47a4-c380-439f',
+                name: 'Container 2',
+                type: 'Container'
+            }
+        ];
+    }));
+
+    it('should change Page', fakeAsync(() => {
+        const filter = 'filter';
+        const page = 1;
+
+        fixture.detectChanges();
+
+        const paginatorService: PaginatorService = de.injector.get(PaginatorService);
+        paginatorService.totalRecords = 2;
+        spyOn(paginatorService, 'getWithOffset').and.returnValue(Observable.of([]));
+
+        fixture.detectChanges();
+
+        this.searchableDropdownComponent.pageChange.emit({
+            filter: filter,
+            first: 10,
+            page: page,
+            pageCount: 10,
+            rows: 0
+        });
+
+        tick();
+        expect(paginatorService.getWithOffset).toHaveBeenCalledWith(10);
+    }));
+
+    it('should paginate when the filter change', fakeAsync(() => {
+        const filter = 'filter';
+
+        fixture.detectChanges();
+
+        const paginatorService: PaginatorService = de.injector.get(PaginatorService);
+        paginatorService.totalRecords = 2;
+        spyOn(paginatorService, 'getWithOffset').and.returnValue(Observable.of([]));
+
+        fixture.detectChanges();
+
+        this.searchableDropdownComponent.filterChange.emit(filter);
+        comp.handleFilterChange(filter);
```

Comment:
You should not need call the handleFilterChange method here because is it the callback for 'filterChange.emit'